src/main/java/org/technologybrewery/helmdependency/UploadDependenciesMojo.java

```diff
@@ -67,14 +68,19 @@ public void execute() throws MojoExecutionException {
         credentials.logout();
     }
 
-    private void deployDockerDependentDockerImages(String chartDirectory) throws MojoExecutionException, InterruptedException {
+    private void deployDockerDependentDockerImages(String chartDirectory) throws InterruptedException, MojoExecutionException {
         String chartTarballs = Paths.get(chartDirectory, "target", "helm", "repo").toString();
-        List<String> dockerImages = HelmDependencyUtils.getDependentDockerImages(helm(), chartTarballs);
+        HashSet<String> dockerImages = new HashSet<>(HelmDependencyUtils.getDependentDockerImages(helm(), chartTarballs));
         DockerCommandExecutor executor = new DockerCommandExecutor(project.getBasedir());
         for(String dockerImage : dockerImages) {
             String newTag = String.format("%s%s", dockerUrl, dockerImage);
-            executor.executeAndLogOutput(Arrays.asList("tag", dockerImage, newTag));
-            executor.executeAndLogOutput(Arrays.asList("push", newTag));
+            try {
+                executor.executeAndLogOutput(Arrays.asList("tag", dockerImage, newTag));
+                executor.executeAndLogOutput(Arrays.asList("push", newTag));
+            }
+            catch (Exception e) {
+                getLog().error(e.getMessage());
```

Comment:
Does this not still throw an exception and stop the execution? Would think `.error()` doesn't allow you to keep pushing images?

Reply:
Its just logging the message as an error, it keeps on pushing.